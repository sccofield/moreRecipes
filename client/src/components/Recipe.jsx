import React from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Recipe = props => (
  <div>
    {
      props.recipe &&

      <div className="card">
        <div className="view overlay hm-white-slight image-width">
          <img
            src={props.recipe.imageUrl}
            className="img-fluid image-width"
            alt=""
          />
          <a>
            <div className="mask" />
          </a>
        </div>
        {/* <a className="btn-floating btn-action" href="#">
          <i className="fa fa-heart-o" />
        </a> */}
        <div className="card-body">
          <Link to={`recipes/${props.recipe.id}`}>
            <h4 className="card-title">{props.recipe.title}</h4>
            <hr />
          </Link>
          <p className="card-text">
            {props.recipe.description.slice(0, 140).concat(' ...')}
          </p>

          <p className="author">By
            <span> {props.recipe.userName}</span>
          </p>
        </div>

        <div className="card-data">
          <ul>
            <li>
              <i className="fa fa-clock-o" />
              {moment(props.recipe.createdAt).format('ll')}
            </li>
            <li>

              <i className="fa fa-comments-o" /> {props.recipe.Reviews.length}
            </li>
            <li>
              <i className="fa fa-thumbs-up" /> {props.recipe.Upvotes.length}
            </li>
            <li>
              <i className="fa fa-thumbs-down" />
              {props.recipe.Downvotes.length}
            </li>

          </ul>
        </div>

      </div>
    }
    {
      !props.recipe &&
      <h1>No Recipe added yet ...</h1>
    }
  </div>
);

Recipe.propTypes = {
  recipe: PropTypes.shape({
    imageUrl: PropTypes.string,
    id: PropTypes.number,
    title: PropTypes.string,
    description: PropTypes.string,
    userName: PropTypes.string,
    createdAt: PropTypes.string,
    Reviews: PropTypes.array,
    Downvotes: PropTypes.array,
    Upvotes: PropTypes.array
  })
};

Recipe.defaultProps = {
  recipe: null
};

export default Recipe;
