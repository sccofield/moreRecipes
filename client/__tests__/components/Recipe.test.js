import React from 'react';
import toJSON from 'enzyme-to-json';
import Recipe from '../../src/components/Recipe';

describe('Recipe component', () => {
  const props = {
    recipe: {
      imageUrl: 'image url',
      id: 3,
      title: 'recipe title',
      description: 'recipe description',
      userName: 'username',
      createdAt: 'created at',
      Reviews: [{}],
      Downvotes: [{}],
      Upvotes: [{}]
    }
  };

  it('should render recipe card when props is given', () => {
    const wrapper = shallow(<Recipe {...props} />);
    expect(wrapper.find('div').length).toEqual(6);
    expect(wrapper.find('.card-title').text()).toBe('recipe title')
    expect(wrapper.find('.card-text').text()).toBe('recipe description ...')
      
    
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('should render text when props is not given', () => {
    const wrapper = shallow(<Recipe />);
    expect(wrapper.find('h1').length).toEqual(1);
    expect(wrapper.find('h1').text()).toBe('No Recipe added yet ...')
  })
});

