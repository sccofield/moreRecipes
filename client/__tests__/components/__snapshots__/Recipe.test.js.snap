// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe component should render recipe card when props is given 1`] = `
<div>
  <div
    className="card"
  >
    <div
      className="view overlay hm-white-slight image-width"
    >
      <img
        alt=""
        className="img-fluid image-width"
        src="image url"
      />
      <a>
        <div
          className="mask"
        />
      </a>
    </div>
    <div
      className="card-body"
    >
      <Link
        replace={false}
        to="recipes/3"
      >
        <h4
          className="card-title"
        >
          recipe title
        </h4>
        <hr />
      </Link>
      <p
        className="card-text"
      >
        recipe description ...
      </p>
      <p
        className="author"
      >
        By
        <span>
           
          username
        </span>
      </p>
    </div>
    <div
      className="card-data"
    >
      <ul>
        <li>
          <i
            className="fa fa-clock-o"
          />
          Invalid date
        </li>
        <li>
          <i
            className="fa fa-comments-o"
          />
           
          1
        </li>
        <li>
          <i
            className="fa fa-thumbs-up"
          />
           
          1
        </li>
        <li>
          <i
            className="fa fa-thumbs-down"
          />
          1
        </li>
      </ul>
    </div>
  </div>
</div>
`;
